bundler/lib/bundler/remote_specification.rb

```diff
@@ -97,12 +97,48 @@ def git_version
       " #{source.revision[0..6]}"
     end
 
+    # we don't get the checksum from a server like we could with EndpointSpecs
+    # calculating the checksum from the file on disk still provides some measure of security
+    # if it changes from install to install, that is cause for concern
+    def checksum
+      @checksum ||= begin
+        return unless _fetched_gem
+
+        File.open(_fetched_gem) do |f|
+          digest = Bundler::SharedHelpers.digest(:SHA256).new
+          digest << f.read(16_384) until f.eof?
+
+          hexdigest = digest.hexdigest!
+          hexdigest
+        end
+      end
+    end
+
+    def to_checksum
+      digest = "sha256-#{checksum}" if checksum
+      Bundler::Checksum.new(name, version, platform, digest)
+    end
+
     private
 
     def to_ary
       nil
     end
 
+    def _fetched_gem
+      @_fetched_gem ||= begin
```

Comment:
are you concerned about keeping the entire gem contents in memory if it's cached on the remote spec?

Reply:
Once I updated to use `download_gem`, I don't think this is a problem anymore